src/daft-csv/src/read.rs

```diff
@@ -328,7 +325,11 @@ async fn read_csv_single_into_table(
 
     let filtered_tables = tables.map_ok(move |table| {
         if let Some(predicate) = &predicate {
-            let filtered = table?.filter(&[predicate.clone()])?;
+            let table = table?;
+
+            let predicate = BoundExpr::try_new(predicate.clone(), &table.schema)?;
```

Comment:
Yes haha just saw this! Gotta bind in the table schema rather than the output schema. Approved 🙌